extensions/openapi-generator/codegen/src/main/java/io/quarkus/openapi/generator/deployment/OpenApiGeneratorCodeGen.java

```diff
@@ -0,0 +1,78 @@
+package io.quarkus.openapi.generator.deployment;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import io.quarkus.bootstrap.prebuild.CodeGenException;
+import io.quarkus.deployment.CodeGenContext;
+import io.quarkus.deployment.CodeGenProvider;
+import io.quarkus.openapi.generator.deployment.wrapper.OpenApiClientGeneratorWrapper;
+import io.quarkus.utilities.OS;
+
+/**
+ * Code generation for OpenApi Client. Generates Java classes from OpenApi spec files located in src/main/openapi or
+ * src/test/openapi
+ * <p>
+ * Wraps the <a href="https://openapi-generator.tech/docs/generators/java">OpenAPI Generator Client for Java</a>
+ */
+public class OpenApiGeneratorCodeGen implements CodeGenProvider {
+
+    private static final String YAML = ".yaml";
+    private static final String YML = ".yml";
+    private static final String JSON = ".json";
+
+    @Override
+    public String providerId() {
+        return "openapi-generator";
+    }
+
+    @Override
+    public String inputExtension() {
+        return "json";
+    }
+
+    @Override
+    public String inputDirectory() {
+        return "openapi";
+    }
+
+    @Override
+    public boolean trigger(CodeGenContext context) throws CodeGenException {
+        // TODO: skip codegen?
+        final Path outDir = context.outDir();
+        final Path openApiDir = context.inputDir();
+        try {
+            if (Files.isDirectory(openApiDir)) {
+                try (Stream<Path> openApiFilesPaths = Files.walk(openApiDir)) {
+                    final List<String> openApiFiles = openApiFilesPaths
+                            .filter(Files::isRegularFile)
+                            .map(Path::toString)
+                            .filter(s -> s.endsWith(YAML) || s.endsWith(YML) || s.endsWith(JSON))
```

Comment:
With `inputExtension` returning only `json`, `trigger()` won't be called on `*.yaml` and `*.yml` file changes. 
I think it would be good to change `inputExtension` to a collection or array. I think it should be simple, would require propagating it and changing https://github.com/quarkusio/quarkus/blob/main/core/deployment/src/main/java/io/quarkus/deployment/util/FSWatchUtil.java#L51 to `contains` instead of `equals`

Reply:
an alternative is to do what we have for Avro, separate `CodeGenProvider` for each file extension

Reply:
Yeah, I had this feeling... well, I'll do the Acro way for now ;)